lib/new_relic/agent/instrumentation/mongo.rb

```diff
@@ -16,7 +16,44 @@
 
   executes do
     NewRelic::Agent.logger.info 'Installing Mongo instrumentation'
-    install_mongo_instrumentation
+    if NewRelic::Agent::Datastores::Mongo.is_monitoring_enabled?
+      install_mongo_command_subscriber
+    else
+      install_mongo_instrumentation
+    end
+  end
+
+  class CommandSubscriber
+
+    def started(event)
+      operations[event.operation_id] = event
+    end
+
+    def succeeded(event)
+      started_event = operations.delete(event.operation_id)
+      NewRelic::Agent.instance.transaction_sampler.notice_nosql_statement(
+        format_statement(started_event.command), event.duration
+      )
```

Comment:
There's some good samples of the steps you'd want to actually take to record over in https://github.com/newrelic/rpm/blob/master/lib/new_relic/agent/instrumentation/active_record_subscriber.rb#L53-L83. You would want to use `NewRelic::Agent::Datastore::MetricHelper.metrics_for` instead of the ActiveRecord flavor, but overall the steps would be very similar.